src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/MixinWorld.java

```diff
@@ -115,4 +123,26 @@ public IBigCube getCube(int cubeX, int cubeY, int cubeZ, ChunkStatus requiredSta
             return icube;
         }
     }
+
+    @Inject(method = "isLoaded", at = @At("HEAD"), cancellable = true)
+    private void isLoaded3D(BlockPos pos, CallbackInfoReturnable<Boolean> cir) {
+        if (!this.isCubic) {
+            return;
+        }
+        if (this.isOutsideBuildHeight(pos)) {
+            cir.setReturnValue(false);
+        }
+
+        cir.setReturnValue(((ICubeProvider) this.getChunkSource()).hasCube(Coords.blockToCube(pos.getX()), Coords.blockToCube(pos.getY()), Coords.blockToCube(pos.getZ())));
+    }
+
+    @SuppressWarnings("ConstantConditions") @Redirect(method = "loadedAndEntityCanStandOnFace",
+        at = @At(value = "INVOKE", target = "Lnet/minecraft/world/level/Level;getChunk(IILnet/minecraft/world/level/chunk/ChunkStatus;Z)Lnet/minecraft/world/level/chunk/ChunkAccess;"))
```

Comment:
Could this be done in an `Inject` at HEAD instead of `Redirect`? That way we don't have to duplicate vanilla code in the else branch.

Reply:
Nope, we'd actually be copying a lot more vanilla code in this case, and we want to purposely break here if some other mod redirects otherwise